HelloImKevo/SkeletonPaymentApp-Senior
# [Room] Upgrade to version 2.5.1 to fix DAO Cursor conversion compiler errors

Summary:
It's been a while since I opened this project, but to my dismay, it no longer
compiles correctly in Android Studio Dolphin+ due to these errors:
```
Android/app/src/main/java/com/imobile3/groovypayments/data/dao/CartDao.java:24:
error: Not sure how to convert a Cursor to this method's return type
(java.util.List<com.imobile3.groovypayments.data.model.Cart>).
    List<Cart> getCarts();

Android/app/src/main/java/com/imobile3/groovypayments/data/dao/ProductDao.java:24:
error: Not sure how to convert a Cursor to this method's return type
(java.util.List<com.imobile3.groovypayments.data.model.Product>).
    List<Product> getProducts();
```

The only way I could fix the issue was by upgrading the Room version to
something more recent, like 2.5.1 which resolved those two compiler errors, but
then introduced new compiler errors due to the way the `@ForeignKey`
implementation changed. The Foreign Keys are now declared at the Entity
class level. Proper app functionality should be restored now.

Additionally, several IDE-specific XML files have been added to the `.idea`
folder - not sure if those should be gitignored. We'll find out in subsequent
efforts!

Further reading:
https://developer.android.com/training/data-storage/room/
https://stackoverflow.com/questions/70397235/room-foreignkey-this-annotation-is-not-applicable-to-target-member-property-wit

Test Plan:
- Run Gradle Sync and build the Project.
- Confirm that there are no compiler errors on Android Studio Dolphin+.
- Deploy the Groovy app to a Nexus API 27 emulator.
- Navigate to the "Order Entry" screen.
- Confirm that you can add a product to the cart without issue.
